GridTools-Glossary.md

```diff
@@ -14,11 +16,27 @@ Describes a set of [grid points](#grid-point) on which the computation is applie
 
 ### Halo
 
+### ICON
+
+A weather and climate model developed by DWD and MPI-M build on a icosahedral grid. ICON uses an [unstructured](#Unstructured-Grid), but the grid can also be represented as partially [structured](#Structured-grid), see [Pentagon problem](#pentagon-problem).
+
 ### Iteration Space
 
+TODO disambiguate from compute domain or make it an alias.
+
 ### Origin
 
-### Unstructured
+### Pentagon problem
+
+In the ICON grid, the sphere is built from rhomboids. Within each rhomboid the mesh and dual-mesh are fully structured (triangles, each vertex has exactly 6 edges). At corner points where the rhomboids are patched together, each vertex has only 5 neighbors. TODO add picture.
+
+### Structured Grid
+
+A grid where cells follow a regular pattern that allows strided access to neighbors. Examples of structured grids are [Cartesian grids](#Cartesian-Grid) or the structured representations of the ICON grid.
```

Comment:
I think the structured / unstructured grid definitions should be rephrased to clarify that the property of regularity/irregularity is related to the connectivity of the mesh, and not to the actual positions of the vertices to avoid misunderstandings. In my opinion, Wikipedia shows a good example on how this could be expressed (https://en.wikipedia.org/wiki/Types_of_mesh#Classification_of_grids) and also some nice picture examples for 2d meshes in the article about 2d regular tilings (https://en.wikipedia.org/wiki/Euclidean_tilings_by_convex_regular_polygons#Regular_tilings).

Reply:
After looking at examples of 2d semiregular tilings in the previous link, I now have the doubt if the 2d semiregular tilings should be also considered as structured grids even if they have different cell types, since in principle they follow a regular pattern...